frontend/src/components/DropGem/DropGem.vue

```diff
@@ -0,0 +1,43 @@
+<template>
+  <NavBar v-bind:left-arrow="currentStageIdx > 0" @click-left="prevStage" />
+  <router-view class="select-friend" v-bind:name="currentStage"></router-view>
+  <div>
+    <button v-if="currentStageIdx < stages.length - 1" v-on:click="nextStage">
+      Next
+    </button>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent } from "vue";
+import { NavBar } from "vant";
+
+export default defineComponent({
+  data() {
+    return {
+      stages: ["friend", "media", "message"],
+      currentStageIdx: 0,
```

Comment:
I think we could replace this with a Typescript enum, and then just have a `currentStage` which is an instance of that enum. Iirc, you can add 1 / minus 1 to that enum as well. Not 100% certain though, so feel free to ignore if it doesn't work

https://www.typescriptlang.org/docs/handbook/enums.html

Reply:
Updated, thanks for the suggestion!